pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/LLCSegmentCompletionHandlers.java

```diff
@@ -416,166 +387,120 @@ public String segmentCommitEndWithMetadata(@QueryParam(SegmentCompletionProtocol
   }
 
   /**
-   * Extract and return the segment metadata from the two input form data files (metadata file and creation meta).
-   * Return null if any of the two files is missing or there is exception during parsing and extraction.
+   * Extracts the segment file from the form into a local temporary file under file upload temporary directory.
    */
-  private SegmentMetadataImpl extractMetadataFromInput(FormDataMultiPart metadataFiles, String segmentNameStr) {
-    String tempMetadataDirStr = StringUtil.join("/", _controllerConf.getLocalTempDir(),
-        segmentNameStr + METADATA_TEMP_DIR_SUFFIX + String.valueOf(System.currentTimeMillis()));
-    File tempMetadataDir = new File(tempMetadataDirStr);
+  private static File extractSegmentFromFormToLocalTempFile(FormDataMultiPart form, String segmentName)
+      throws IOException {
     try {
-      Preconditions.checkState(tempMetadataDir.mkdirs(), "Failed to create directory: %s", tempMetadataDirStr);
-      // Extract metadata.properties from the metadataFiles.
-      if (!extractMetadataFromInputField(metadataFiles, tempMetadataDirStr, V1Constants.MetadataKeys.METADATA_FILE_NAME,
-          segmentNameStr)) {
-        return null;
-      }
-      // Extract creation.meta from the metadataFiles.
-      if (!extractMetadataFromInputField(metadataFiles, tempMetadataDirStr, V1Constants.SEGMENT_CREATION_META,
-          segmentNameStr)) {
-        return null;
+      Map<String, List<FormDataBodyPart>> map = form.getFields();
+      Preconditions.checkState(PinotSegmentUploadDownloadRestletResource.validateMultiPart(map, segmentName),
```

Comment:
can you add the instance name to this log?

Reply:
Instance name will be logged by the caller of this method